src/proxy/outbound.rs

```diff
@@ -381,35 +382,72 @@ impl OutboundConnection {
             });
         }
 
-        let us = us.unwrap();
+        let mut mutable_us = us.unwrap();
+        let workload_ip = self
+            .pi
+            .state
+            .load_balance(
+                &mutable_us.workload,
+                &source_workload,
+                &self.pi.cfg.dns_resolver_config,
+                &self.pi.cfg.dns_resolver_opts,
```

Comment:
nit: can we make these inner fields of state rather than passing

Reply:
addressed in https://github.com/istio/ztunnel/pull/612/commits/c8615127acf2d41d00812688db412f1d69d514fc